src/service/fixed-layer.js

```diff
@@ -530,44 +528,28 @@ export class FixedLayer {
    * Removes element from the fixed layer.
    *
    * @param {!Element} element
-   * @return {!Array<!ElementDef>}
+   * @return {!ElementDef|undefined}
    * @private
    */
   removeElement_(element) {
-    const removed = [];
     for (let i = 0; i < this.elements_.length; i++) {
       const fe = this.elements_[i];
       if (fe.element == element) {
         this.vsync_.mutate(() => {
           setStyle(element, 'top', '');
         });
         this.elements_.splice(i, 1);
-        removed.push(fe);
+        return fe;
```

Comment:
Do we prevent an element from being added more than once? Looking at `setupElement_` it doesn't seem like it.

Reply:
Yes, we attempt to [find the element's `fe`](https://github.com/ampproject/amphtml/blob/33f26f53dbe0bfc3091e2b2fed34199dfa2e5ef6/src/service/fixed-layer.js#L491-L497). If it's found, we [update](https://github.com/ampproject/amphtml/blob/33f26f53dbe0bfc3091e2b2fed34199dfa2e5ef6/src/service/fixed-layer.js#L499-L503), else we [create a new one](https://github.com/ampproject/amphtml/blob/33f26f53dbe0bfc3091e2b2fed34199dfa2e5ef6/src/service/fixed-layer.js#L505-L521).

Reply:
https://github.com/ampproject/amphtml/blob/33f26f53dbe0bfc3091e2b2fed34199dfa2e5ef6/src/service/fixed-layer.js#L493

Does adding a single element twice with different `position` result in two `fe`s?

Reply:
Undid these changes.